### ISSUE TITLE
Race condition during blob publishing

### ISSUE THREAD

**Issue by JohnTortugo** [dotnet/arcade#4298]
There is the possibility for a race condition during blob publishing when a build that activates two different channels that target the same Sleet feed. The race condition itself isn't the problem. The thing is that we want to make sure that whenever we attempt to re-publish an asset the published content and the newly one are the same.

Discussed this with @mmitche and @riarenas and possible fix is like so:

- Assume the blob storage doesn't accept overrides.
- Try publishing a blob.
--- If publishing happens everything is fine as there was no previous blob there by assumption.
--- If the blob publishing fail:
-----> Check whether the published blob and the local blob has different contents. If they do show an error otherwise everything is fine.

**Comment by markwilkie** [comment]
Are there scenarios where we intend to use sleet moving forward?

**Comment by riarenas** [comment]
Not moving forward, just for servicing.

**Comment by riarenas** [comment]
Oh, but this talks about blobs. @JohnTortugo can you clarify if you actually mean the sleet feeds, or just publishing blobs to the storage account?

**Comment by markwilkie** [comment]
ping @JohnTortugo

**Comment by MattGal** [comment]
Asked by @markwilkie to comment here:  We have lots of places where we could use code (that lives in a hacky version in Buildtools already, but I think that one downloads the storage file to diff, which is bad for perf)  that does noop when trying to upload an identical blob to another, an error when the contents don't match, and uploads otherwise.  Consider putting this into shared functionality.

**Comment by JohnTortugo** [comment]
Makes sense to me. Would be nice to have at least some links here to these places if you remember off the top of your head.
